src/record/peer-record/index.js

```diff
@@ -0,0 +1,98 @@
+'use strict'
+
+const multiaddr = require('multiaddr')
+const PeerId = require('peer-id')
+const Record = require('libp2p-interfaces/src/record')
+
+const Protobuf = require('./peer-record.proto')
+const {
+  ENVELOPE_DOMAIN_PEER_RECORD,
+  ENVELOPE_PAYLOAD_TYPE_PEER_RECORD
+} = require('./consts')
+
+const arraysAreEqual = (a, b) => a.length === b.length && a.sort().every((item, index) => b[index].equals(item))
```

Comment:
This might be worth adding to a utility file in multiaddr or libp2p-utils. Equality, unique, etc are used in other places.

Reply:
https://github.com/libp2p/js-libp2p-utils/pull/8

I will add the others ones on a better opportunity to not get out of the scope of the PR